src/rules/probit/in.jl

```diff
@@ -19,17 +20,23 @@ end
     p = mean(m_out)
     @assert p >= zero(p) && p <= one(p) "The Probit node only accepts messages on its output with values between 0 and 1."
 
+    T = promote_type(eltype(m_out), eltype(m_in))
+
     # calculate auxiliary variables
     γ = mz/sqrt(1+vz)
 
-    # calculate unnormalized moments of g
-    umom1_g = normcdf(γ)*mz + vz*normpdf(γ)/sqrt(1+vz)
-    umom2_g = 2*mz*umom1_g + (vz - mz^2)*normcdf(γ) - vz^2*γ*normpdf(γ)/(1+vz)
-
-    # calculate moments of posterior
-    mom0_pz = 1 - p + (2*p - 1)*normcdf(γ)
-    mom1_pz = 1/mom0_pz*((1-p)*mz + (2*p-1)*umom1_g)
-    mom2_pz = 1/mom0_pz*((1-p)*(vz + mz^2) + (2*p-1)*umom2_g)
+    # calculate moments of g
+    if γ > 0 && p > 0.5
+        log_mom0_pz = logsumexp([log(1-p), log(2*p-1) + normlogccdf(-γ)])
```

Comment:
I would create allocation-free version of logsumexp if possible. Now it allocates 2-element array every time

Reply:
Would be nice indeed. However, rewriting this function for a variable number of arguments actually increases the number of allocations and slows down performance significantly (~8x slower). If we really want to do this we could create separate functions for different number of arguments, but the `maximum` function requires a tuple/ an array anyway.

Commit (6d8bd11) implements a custom `logsumexp` function that is about 10% faster than the function from `StatsFuns`. Commits (222ac5b) and (0c4a9433) replace the original implementations with this custom `logsumexp` function